--- a/drivers/net/ixgbe/ixgbe_pf.c
+++ b/drivers/net/ixgbe/ixgbe_pf.c
@@ -813,6 +813,10 @@ int ixgbe_pf_host_configure(struct rte_eth_dev *eth_dev)
                if (retval == RTE_PMD_IXGBE_MB_EVENT_PROCEED)
                        retval = ixgbe_set_vf_mc_promisc(dev, vf, msgbuf);
                break;
+       case IXGBE_VF_SET_MACVLAN:
+               if (retval == RTE_PMD_IXGBE_MB_EVENT_PROCEED)
+                       retval = 0;
+               break;
        default:
                PMD_DRV_LOG(DEBUG, "Unhandled Msg %8.8x", (unsigned)msgbuf[0]);
                retval = IXGBE_ERR_MBX;

